src/pal/src/numa/numashim.h

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Enable calling numa functions through shims to make it a soft
+// runtime dependency.
+
+#ifndef __NUMASHIM_H__
+#define __NUMASHIM_H__
+
+#if HAVE_NUMA_H
+
+#include <numa.h>
+#include <numaif.h>
+
+// List of all functions from the numa library that are used
+#define FOR_ALL_NUMA_FUNCTIONS \
+    PER_FUNCTION_BLOCK(numa_available) \
+    PER_FUNCTION_BLOCK(mbind_ptr) \
+    PER_FUNCTION_BLOCK(numa_num_possible_cpus) \
+    PER_FUNCTION_BLOCK(numa_max_node) \
+    PER_FUNCTION_BLOCK(numa_allocate_cpumask) \
+    PER_FUNCTION_BLOCK(numa_node_to_cpus) \
+    PER_FUNCTION_BLOCK(numa_bitmask_weight) \
+    PER_FUNCTION_BLOCK(numa_bitmask_isbitset) \
+    PER_FUNCTION_BLOCK(numa_free_cpumask) \
+    PER_FUNCTION_BLOCK(mbind_ptr)
+
+// Declare pointers to all the used numa functions
+#define PER_FUNCTION_BLOCK(fn) extern decltype(fn)* fn##_ptr;
+FOR_ALL_NUMA_FUNCTIONS
+#undef PER_FUNCTION_BLOCK
+
+// Redefine all calls to numa functions as calls through pointers that are set
+// to the functions of libnuma in the initialization.
+#define numa_available() numa_available()
```

Comment:
These are incorrect except of the mbind. All of them should have the _ptr suffix like the mbind does.
Moreover, the numa_num_possible_cpus is defined as mbind_ptr.